### PR TITLE
fix(editor): stop the camera animation on pointer down so a click stays a click

### LINKED ISSUE #10706: Clicks during a camera animation register as drags, causing the Mobile Chrome focus e2e flake
A click that happens while the camera is animating can turn into a drag, because the camera moving under a stationary pointer counts as pointer movement for drag detection. This has surfaced as a flaky examples e2e test on Mobile Chrome since 2026-09-08 ("Focus › edit->edit, focus stays in the text areas when going from shape-to-shape" in `test-focus.spec.ts`): pressing Tab in a note now selects the new note and pans to it with an animation (#10486), and on a narrow viewport the pan is still running when the test clicks the first note, so editing ends instead of staying focused. On a 1280px desktop viewport the new note is already on screen, no pan happens, and the test passes. Locally on main the test fails roughly 1 in 6 attempts on Mobile Chrome; with #10486 reverted it passes 30/30. Expected: a click during any camera animation (zoom to selection, follow user, pan to adjacent note) behaves as a click, and the test is stable.

## Open questions

1. **Should the fix land in the editor's drag detection (ignore camera-driven pointer moves for all tools), or only in the note shape's Tab behavior (pan without animation)?**
   Unsure; the reporter leaves this to the implementer. The editor-level fix covers every camera animation (zoom to selection, follow user), the note-level one only this path.
2. **Should the e2e test also be hardened (wait for the camera to settle, click the note element instead of the body) even after the product fix?**
   Yes.

Confidence: 86%, ready to get started.

### PR DESCRIPTION
**Pull request by MitjaBezensek**
This PR fixes a bug where a click made while the camera is animating registered as a drag. Closes #10706. Since #10486, pressing Tab in a note pans to the new note, and on a narrow viewport clicking during that pan ended editing and moved the note instead of moving focus into it, which is the Mobile Chrome flake in `test-focus.spec.ts`. Supersedes #10711, which re-anchored the drag origin instead.

### Before

`_setCamera` dispatches a synthetic `pointer_move` after every camera write. The drag threshold compares `originPagePoint` to `currentPagePoint`, and a camera moving under a stationary pointer shifts the current page point, so a held-still pointer crossed the threshold and the state chart went to `translating` or `brushing`. Wheel, pinch, middle-mouse pan, and right-click pan already called `stopCameraAnimation()` before moving the camera, but a plain left `pointer_down` did not, so zoom-to-selection, zoom-to-fit, the Tab pan, and a fling that was still sliding all kept running underneath the click.

### After

`pointer_down` calls `stopCameraAnimation()`, which cancels the viewport animation or slide. The camera stays where it was when the pointer went down, the synthetic move measures zero travel, and the click stays a click. A real drag started afterwards is measured from where the pointer actually is. As a side effect, touching the canvas during a fling stops the momentum, which matches native scroll views.

### Implementation notes

Following a user is not stopped on `pointer_down`: it is not a camera animation in the `stop-camera-animation` sense, and stopping it on every tap would break follow mode on touch. A click while following can still drift for the same reason. An animation that starts after the pointer is already down (a keyboard shortcut while holding the mouse) is also not covered. Both are left as is.

The e2e test now waits for the camera to settle, clicks the first note by its `data-shape-id`, and asserts that note's editor is the focused one. Even with the fix, the `body` click was timing dependent: its centre only overlaps the first note before the pan has progressed, and afterwards it lands on the new note. Shape DOM order is not creation order either, so `.tl-shape` `.first()` is the wrong target, and the old any-shape focus assertion let a click on the new note pass.

### Change type

- [x] `bugfix`

### Test plan

1. In the browser console run `editor.user.updateUserPreferences({ animationSpeed: 0.05 })` so the animation is long enough to click into.
2. Draw a filled rectangle, deselect, press shift+1, and mouse-down on the rectangle mid-animation without moving. On `main` the rectangle jumps by the pan distance; with this change the camera stops and the rectangle is selected.
3. Create a note near the right edge of the window so the Tab-created note lands off screen, type, press Tab, and click the first note during the pan. On `main` editing ends and the note is dragged; with this change editing moves into the first note.

- [x] Unit tests — 6 of the 7 new tests in `dragDuringCameraAnimation.test.ts` fail on `main` and all pass with this change
- [x] End to end tests — `test-focus.spec.ts` hardened; 100/100 on Mobile Chrome before the id-targeting change, 25/25 on both Mobile Chrome and chromium after

### Release notes

- Fix a click during a camera animation or fling being treated as a drag

### Code changes

| Section   | LOC change |
| --------- | ---------- |
| Core code | +4 / -0    |
| Tests     | +125 / -5  |

### CODE REVIEW

File: apps/examples/e2e/tests/test-focus.spec.ts [hunk 1]
```diff
@@ -214,8 +215,13 @@ test.describe('Focus', () => {
...
 		// create new note next to it
 		await page.keyboard.press('Tab')
+		await expect(page.locator('.tl-shape')).toHaveCount(2)
 
-		await (await page.$('body'))?.click()
+		// Tab pans to the new note; let the camera settle so the first note is where
+		// the locator resolves it before clicking.
+		await page.waitForFunction(() => editor.getCameraState() === 'idle')
+
+		await page.locator('.tl-shape').first().click()
```
[the remark this answers is not in the document]
Comment (MitjaBezensek) [reply apps/examples/e2e/tests/test-focus.spec.ts]: Fixed in a followup commit.

**Review by kaneel — APPROVED** [review]
It's s an edge case but it checks out.
